packages/core/src/modules/drpc/__tests__/drpc-messages.e2e.test.ts

```diff
@@ -0,0 +1,199 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import type { SubjectMessage } from '../../../../../../tests/transport/SubjectInboundTransport'
+import type { ConnectionRecord } from '../../connections'
+import type { DRPCResponseObject } from '../messages'
+
+import { Subject } from 'rxjs'
+
+import { SubjectInboundTransport } from '../../../../../../tests/transport/SubjectInboundTransport'
+import { SubjectOutboundTransport } from '../../../../../../tests/transport/SubjectOutboundTransport'
+import { getInMemoryAgentOptions, makeConnection } from '../../../../tests/helpers'
+import testLogger from '../../../../tests/logger'
+import { Agent } from '../../../agent/Agent'
+import { DRPCErrorCode } from '../messages'
+
+const faberConfig = getInMemoryAgentOptions('Faber DRPC Messages', {
+  endpoints: ['rxjs:faber'],
+})
+
+const aliceConfig = getInMemoryAgentOptions('Alice DRPC Messages', {
+  endpoints: ['rxjs:alice'],
+})
+
+describe('DRPC Messages E2E', () => {
+  let faberAgent: Agent
+  let aliceAgent: Agent
+  let faberConnection: ConnectionRecord
+  let aliceConnection: ConnectionRecord
+
+  beforeEach(async () => {
+    const faberMessages = new Subject<SubjectMessage>()
+    const aliceMessages = new Subject<SubjectMessage>()
+    const subjectMap = {
+      'rxjs:faber': faberMessages,
+      'rxjs:alice': aliceMessages,
+    }
+
+    faberAgent = new Agent(faberConfig)
+    faberAgent.registerInboundTransport(new SubjectInboundTransport(faberMessages))
+    faberAgent.registerOutboundTransport(new SubjectOutboundTransport(subjectMap))
+    await faberAgent.initialize()
+
+    aliceAgent = new Agent(aliceConfig)
+    aliceAgent.registerInboundTransport(new SubjectInboundTransport(aliceMessages))
+    aliceAgent.registerOutboundTransport(new SubjectOutboundTransport(subjectMap))
+    await aliceAgent.initialize()
+    ;[aliceConnection, faberConnection] = await makeConnection(aliceAgent, faberAgent)
+  })
+
+  afterEach(async () => {
+    await faberAgent.shutdown()
+    await faberAgent.wallet.delete()
+    await aliceAgent.shutdown()
+    await aliceAgent.wallet.delete()
+  })
+
+  test('Alice and Faber exchange messages', async () => {
+    testLogger.test('Alice sends message to Faber')
+    faberAgent.drpcMessages.createDRPCMethodHandler('hello', async (message) => {
```

Comment:
I may have interpreted the RFC differently, or maybe this is how it's intended to work in `credo-ts`, but in ACA-Py the agent doesn't directly handle the RPC-JSON object (i.e. it doesn't handle the RPC method execution, it simply forwards that through to a controller. The controller is responsible for handling the RPC message accordingly and formulating a response. So, in this case, would an application running `credo-ts` have to register a method handler directly with the agent in order to use DRPC?

I'm quoting this line from the [Key Concepts](https://github.com/hyperledger/aries-rfcs/blob/bfa0bb23bda457a92a2a37a33b396f43ae84eb03/features/0804-didcomm-rpc/README.md#key-concepts) of RFC 0804:

>... The interpretation of the request, how to carry out the request, the content of the response, and the interpretation of the response, are all up to the business logic (controllers) of the participating agents...